pkg/etcd/migrate.go

```diff
@@ -0,0 +1,235 @@
+				return errors.Trace(err)
+			}
+		}
+	}
+
+	// 5. update meataVersion
+	_, err = m.cli.Put(ctx, m.metaVersionKey, fmt.Sprintf("%d", m.newMetaVersion))
+	if err != nil {
+		log.Error("update meta version failed, etcd meta data migration failed", zap.Error(err))
+		return errors.Trace(err)
+	}
+
+	log.Info("etcd data migration successful")
+	return nil
+}
+
+func (m migrater) campaignOldOwner(ctx context.Context) error {
+	sess, err := concurrency.NewSession(m.cli.Unwrap(),
+		concurrency.WithTTL(etcdSessionTTL))
+	if err != nil {
+		return errors.Trace(err)
+	}
+	election := concurrency.NewElection(sess, m.oldOwnerKey)
+	defer func() {
+		_ = sess.Close()
+	}()
+
+	// TODO(dongmen): do more investigration about how to handle error here
+	if err := election.Campaign(ctx, migrationCampaignKey); err != nil {
+		switch errors.Cause(err) {
+		case context.Canceled, mvcc.ErrCompacted:
+		default:
+			// if campaign owner failed, restart capture
+			log.Warn("campaign owner failed", zap.Error(err))
+		}
+		return errors.Trace(err)
+	}
+	return nil
+}
+
+func MigrateData(ctx context.Context, cli *Client) error {
+	version, err := getMetaVersion(ctx, cli)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	shouldMigrate := false
+	oldVersion, newVersion := 0, cdcMetaVersion
+
+	if version == noMetaVersion {
+		shouldMigrate = true
+	} else {
+		oldVersion = version
+		shouldMigrate = oldVersion < newVersion
+	}
+
+	if !shouldMigrate {
+		return nil
+	}
+
+	m := migrater{
+		oldMetaVersion:    oldVersion,
+		newMetaVersion:    cdcMetaVersion,
+		metaVersionKey:    DefaultClusterAndMetaPrefix + metaVersionKey,
+		oldOwnerKey:       "/ticdc/cdc/owner",
+		cli:               cli,
+		keyPrefixs:        make(keys),
+		etcdNoMetaVersion: version == noMetaVersion,
+	}
+
+	m.keyPrefixs.addPair("/tidb/cdc/changefeed/info", DefaultClusterAndNamespacePrefix+changefeedInfoKey)
+	m.keyPrefixs.addPair("/tidb/cdc/job", DefaultClusterAndNamespacePrefix+changefeedStatusKey)
+
+	return m.migrate(ctx)
+}
+
+func ShouldMigrate(ctx context.Context, cli *Client) (bool, error) {
+	version, err := getMetaVersion(ctx, cli)
+	if err != nil {
+		return false, errors.Trace(err)
+	}
+	return version != cdcMetaVersion, nil
+}
+
+// WaitMetaVersionMatched checks and waits until the metaVersion in etcd
+// matched to loack cdcMetaVersion
+func WaitMetaVersionMatched(ctx context.Context, cli *Client) error {
+	tick := time.NewTicker(time.Second)
+	version, err := getMetaVersion(ctx, cli)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	if version == cdcMetaVersion {
+		return nil
+	}
+
+	for {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		case <-tick.C:
+			version, err := getMetaVersion(ctx, cli)
+			if err != nil {
+				return errors.Trace(err)
+			}
+			if version == cdcMetaVersion {
+				return nil
+			}
+		}
+	}
+}
+
+func getMetaVersion(ctx context.Context, cli *Client) (int, error) {
+	key := DefaultClusterAndMetaPrefix + metaVersionKey
```

Comment:
we can get the cluster id from cmd parameter, so we should not use `DefaultClusterAndMetaPrefix`, although the cluster id is "default"